src/jit/morph.cpp

```diff
@@ -10327,10 +10327,10 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
                         // IF we get here we should be changing 'oper' 
                         assert(tree->OperGet() != oper);
 
-                        ValueNumPair vnp;
-                        vnp = tree->gtVNPair;  // Save the existing ValueNumber for 'tree'
-
-                        tree->SetOper(oper);
+                        // Keep the old ValueNumber for 'tree' as the new expr
+                        // will still compute the same value as before
+                        bool clearVN = false;
+                        tree->SetOper(oper, clearVN);
```

Comment:
An option here to improve readability is to use a user defined enum like this:

enum ValueNumberUpdate = {CLEAR_VN, PRESERVE_VN);

Instead of using a bool and passing true or false.